components/FileInput.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import { useDropzone, DropzoneProps } from 'react-dropzone';
+import { Controller, Control } from 'react-hook-form';
+import Image from 'next/image';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+
+interface FileInputProps extends DropzoneProps {
+  name: string;
+  instructions: string;
+  selectedFileMessage: string;
+  control: Control;
+  rules?: object;
+}
+
+const FileInput = ({
+  name,
+  instructions,
+  selectedFileMessage,
+  control,
+  accept,
+  maxSize,
+  minSize,
+  multiple,
+  rules,
+}: FileInputProps) => {
+  const [
+    filePreview,
+    setFilePreview,
+  ] = useState<string | null>(null);
+
+  return (
+    <Controller
+      name={name}
+      control={control}
+      rules={rules}
+      render={({ field: { onChange }, fieldState: { error } }) => {
+        const handleDrop = (acceptedFiles: File[]) => {
+          onChange(acceptedFiles.length > 0 ? acceptedFiles[0] : null);
+          if (acceptedFiles[0] && acceptedFiles[0].type.startsWith('image/')) {
+            setFilePreview(URL.createObjectURL(acceptedFiles[0]));
```

Comment:
It's a good practice to call [revokeObjectUrl](https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL_static) in order to release previous images from memory upon dropping a new file or when you unmount this file input.